src/main/java/edu/umich/eecs/april/apriltag/TagView.java

```diff
@@ -505,7 +506,23 @@ private void setHighestCameraPreviewResolution(Camera camera)
         parameters.setPreviewSize(bestSize.width, bestSize.height);
         Log.i(TAG, "Setting " + bestSize.width + " x " + bestSize.height);
 
-        parameters.setFocusMode(Camera.Parameters.FOCUS_MODE_CONTINUOUS_VIDEO);
+        List<String> focusModes = parameters.getSupportedFocusModes();
+        if (focusModes.contains(Camera.Parameters.FOCUS_MODE_CONTINUOUS_VIDEO)) {
+            parameters.setFocusMode(Camera.Parameters.FOCUS_MODE_CONTINUOUS_VIDEO);
+            Log.i(TAG, "Setting focus mode for continuous video");
+        } else {
+            Log.i(TAG, "Focus mode for continuous video not supported, skipping");
+        }
+
+        List<int[]> fpsRanges = parameters.getSupportedPreviewFpsRange();
+        for (int[] range : fpsRanges) {
+            Log.i(TAG, "[" + range[0] + ", " + range[1] + "]");
+            if (range[0] == 30000 && range[1] == 30000) {
```

Comment:
Just curious, what's the FPS if this doesn't get executed?

Reply:
Unclear, docs suggested it's based on device settings which vary.